test/kubernetes_unittest.cc

```diff
@@ -137,4 +143,47 @@ TEST_F(KubernetesTest, ComputePodAssociations) {
       ComputePodAssociations(reader, pod->As<json::Object>());
   EXPECT_EQ(expected_associations->ToString(), associations->ToString());
 }
+
+TEST_F(KubernetesTest, GetLegacyResource) {
+  Configuration config(std::stringstream(
+    "KubernetesClusterName: TestClusterName\n"
+    "MetadataApiResourceTypeSeparator: \".\"\n"
+    "InstanceZone: TestZone\n"
+    "InstanceId: TestID\n"
+  ));
+  Environment environment(config);
+  KubernetesReader reader(config, nullptr);  // Don't need HealthChecker.
+  json::value pod = json::object({
+    {"metadata", json::object({
+      {"namespace", json::string("TestNamespace")},
+      {"name", json::string("TestName")},
+      {"uid", json::string("TestUid")},
+    })}
+  });
+  const auto m = GetLegacyResource(reader, json::object({
+    {"metadata", json::object({
```

Comment:
Isn't this just `pod->As<json::Object>()`?

Reply:
Removed pod, it was supposed to be inlined.

Reply:
I was actually suggesting you keep `pod`...